### PR TITLE
Add derives for data model

### LINKED ISSUE #1265: Add deserialize and serialize derive to all items in data model
## Add deserialize and serialize derive to all items in iroha_data_model

Iroha-python client depends on serde serialization for pythonize library, thefore requires all data model types to implement serialize and deserialize.

### PR DESCRIPTION
**Pull request by i1i1**
Closes  #1265

### Description of the Change

Adds derive to every item in data model. Also adds schema generation to ci

### CODE REVIEW

File: .github/workflows/iroha2-dev.yml [hunk 1]
```diff
@@ -122,3 +122,22 @@ jobs:
         uses: codecov/codecov-action@v1
         with:
           file: lcov.info
+
+  print-schema:
+    runs-on: [self-hosted, Linux] #ubuntu-latest
+    container: rust:1.52-buster
```
Comment (Revertron) [inline .github/workflows/iroha2-dev.yml:128]: Maybe it's time for 1.54?

**Review by Revertron — DISMISSED** [review]

File: .github/workflows/iroha2-dev.yml (hunk 1 above)
Comment (i1i1) [reply .github/workflows/iroha2-dev.yml:128]: Next pr incoming. Fixing lints of clippy

Comment (eadventurous) [inline .github/workflows/iroha2-dev.yml:128]: I think this should be a step in `Publish artifacts` job.

File: .github/workflows/iroha2-dev.yml [hunk 2]
```diff
@@ -122,3 +122,22 @@ jobs:
...
+          path: ${{ env.CACHE_PATHS }}
+          key: ${{ hashFiles('Cargo.toml') }}
+      - name: Run debug tests and save telemetry
+        run: |
+          mkdir -p target/schema
+          cargo run -p iroha_schema_bin > target/schema/schema.json
+      - name: Upload schema
+        uses: actions/upload-artifact@v2
+        with:
+          name: telemetry
```
Comment (eadventurous) [inline .github/workflows/iroha2-dev.yml:142]:
```suggestion
      - name: Generate schema
        run: |
          mkdir -p target/schema
          cargo run -p iroha_schema_bin > target/schema/schema.json
      - name: Upload schema
        uses: actions/upload-artifact@v2
        with:
          name: schema
```

File: iroha_schema/iroha_schema_bin/src/main.rs [hunk 3]
```diff
@@ -16,13 +16,35 @@ macro_rules! to_json {
...
 
 fn main() {
-    println!(
-        "{}",
-        to_json!(
-            VersionedTransaction,
-            VersionedSignedQueryRequest,
-            VersionedQueryResult,
-            VersionedEventSocketMessage,
-        )
```
Comment (eadventurous) [inline iroha_schema/iroha_schema_bin/src/main.rs:26]: Clients need this, we can't remove it.

File: iroha_schema/iroha_schema_bin/src/main.rs [hunk 4]
```diff
@@ -16,13 +16,35 @@ macro_rules! to_json {
...
-    )
+    // $ rg '^pub (struct|enum)' |cut -d' ' -f3|sed 's/[(< ].*//'
+    let json = to_json! {
+        Add,
+        And,
```
Comment (eadventurous) [inline iroha_schema/iroha_schema_bin/src/main.rs:22]: Is this for python lib?

Comment (eadventurous) [reply iroha_schema/iroha_schema_bin/src/main.rs:22]: Also with the top level structs that were previously here we did get all the ones that you listed here. No need to change this part as far as I can see.

**Review by eadventurous — CHANGES_REQUESTED** [review]

File: iroha_schema/iroha_schema_bin/src/main.rs (hunk 4 above)
Comment (i1i1) [reply iroha_schema/iroha_schema_bin/src/main.rs:22]: I guess we can miss some things. That is why shell one liner is there for regeneration.

Comment (i1i1) [reply iroha_schema/iroha_schema_bin/src/main.rs:22]: No, this is just for schema.

Comment (eadventurous) [reply iroha_schema/iroha_schema_bin/src/main.rs:22]: Check my upper comment. Those 4 structs are exposed in our API, others that are needed for them are just recursively saved.

Comment (eadventurous) [reply iroha_schema/iroha_schema_bin/src/main.rs:22]: We haven't forgotten anything - all our API is implemented in java client lib with current json :)
That's the point of our schema generator design, that we only outline the ones exposed in API.

Comment (i1i1) [reply iroha_schema/iroha_schema_bin/src/main.rs]: Added versioned types also. Kinda feel that it is better for that to be explicit.

Comment (eadventurous) [reply iroha_schema/iroha_schema_bin/src/main.rs]: Why? I mean schema is designed only to expose API. And schema is needed for API :)
I do disagree here. We can also ask Rinat and Roman for their opinions. Maybe have a quick call right now.

**Review by eadventurous — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Review by Revertron — APPROVED** [review]

**Pull request merged**